### DISCUSSION TITLE
smtp connect faild ?! pleas help

### DISCUSSION THREAD

**Discussion by taymazmostafaei** [PHPMailer/PHPMailer#2311] [General]
```
PHPMailer\PHPMailer\Exception: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting in /home/asrebar1/public_html/wp-content/themes/irankala/templates/td-file/PHPMailer-master/PHPMailer.php:1954
Stack trace:
#0 /home/asrebar1/public_html/wp-content/themes/irankala/templates/td-file/PHPMailer-master/PHPMailer.php(1635): PHPMailer\PHPMailer\PHPMailer->smtpSend('Date: Mon, 12 A...', 'This is a multi...')
#1 /home/asrebar1/public_html/wp-content/themes/irankala/templates/td-file/PHPMailer-master/PHPMailer.php(1468): PHPMailer\PHPMailer\PHPMailer->postSend()
#2 /home/asrebar1/public_html/wp-content/themes/irankala/templates/td-file/ajax-contect-us.php(51): PHPMailer\PHPMailer\PHPMailer->send()
#3 {main}
```
i have this problem from this php code 
```
<?php
$obj = json_decode($_POST['obj']) ;

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;
use PHPMailer\PHPMailer\SMTP;

require 'PHPMailer-master/Exception.php';
require 'PHPMailer-master/PHPMailer.php';
require 'PHPMailer-master/SMTP.php';


//Server settings
$mail = new PHPMailer(true) ;
   $mail->isSMTP();
   $mail->Host = "smtp.gmail.com" ;
   $mail->SMTPAuth = true ;
   $mail->SMTPSecure = "ssl" ; //ssl
   $mail->Port = 465 ; //465
   $mail->Username = "turkdevlopers@gmail.com" ;
   $mail->Password = "password" ;                       

//Recipients
$mail->setFrom('turkdevlopers@gmail.com', 'turk devlopers email system');
$mail->addAddress('asrebartar.mobile@gmail.com', 'siyamak'); 
$mail->addAddress('taymazmostafaei3@gmail.com','taymaz mostafaei') ;    //Add a recipient

//Content
$mail->isHTML(true);
$mail->AltBody = 'this is email' ;
$mail->Subject = 'asrebartar contact-us';
$mail->Body    =  '
<div style="direction: rtl;font-family: \'Segoe UI\', Tahoma, Geneva, Verdana, sans-serif;">
<h2 style="color: black;">
سلام پیام جدیدی از ارتباط باما سایتت داری
</h2>
<h2 style="color: black;">
اطلاعات دوست پیام دهنده:
'.$obj->name.''.$obj->lastname.' //
  '.$obj->number.'
</h2>

  <hr>
  <p style="color: black;font-size: 20px;"><b>'.$obj->message.'</b></p>
  <hr>
  </div>';

  //checking for how many time try to send message ## 0 = send warning   1 = Success  3 = try warning  ##
  
      try{
          $mail->send() ;
          echo 1;
      } catch (Exception $e) {
          echo $e;
      }


```

**Comment by DRRSchmidt** [comment]
I have the same problem. See issue #2295. Here is my output from smtp_check.php:
2021-04-14 15:28:43 Connection: opening to SMTP2Go.com:25, timeout=30, options=array() 2021-04-14 15:28:44 Connection failed. Error #2: stream_socket_client(): unable to connect to SMTP2Go.com:25 (Connection refused) [/home/****/php/PHPMailer/src/SMTP.php line 393] 2021-04-14 15:28:44 SMTP ERROR: Failed to connect to server: Connection refused (111) SMTP error: Connect failed
There was no stack trace from this code. The common thread between taymazmostafaei's code and mine (and possibly issue #2295) is that we did not use autoload. I do not have command line access. Please help us find what's missing when we manually install PHPMailer.

**Reply by NiklasBr** [reply]
Your error says:
> SMTP ERROR: Failed to connect to server: Connection refused 

That means that the server you are trying to connect to is not accepting your connection.

Verify your settings and ask the owner of the server if they allow it.

If you are using Gmail, [read the documentation](https://github.com/PHPMailer/PHPMailer/wiki).

**Reply by Synchro** [reply]
This has nothing to do with autoload, you have nothing missing. Your error looks straightforward: "Connection refused" means that you have a network block or other configuration issue preventing you from connecting to your mail server.

**Reply by taymazmostafaei** [reply]
i cant fix it so i didnt used SMTP , and i try to use my server own mail system and it was successful .

**Reply by jabcreations** [reply]
Tay, are you using a VPN? It's very easy for system admins to block IP addresses from spammers who ALSO use VPNs which in turn blocks legitimate users. Also: have you tried a very minimal message so nothing gets flagged?
